### PR TITLE
Gmail OAuth 2.0 authentication

### PR DESCRIPTION
**Pull request by adi88d**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/18538

## Description
Create integration for Gmail using OAuth 2.0 authentication for login, no need for super admin account privileges.

### CODE REVIEW

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.py [hunk 1]
```diff
@@ -0,0 +1,904 @@
...
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+import urllib.parse
+
+''' GLOBAL VARS '''
+params = demisto.params()
+EMAIL = params.get('email', '')
+PROXY = params.get('proxy')
+DISABLE_SSL = params.get('insecure', False)
+FETCH_TIME = params.get('fetch_time', '1 days')
+OPROXY_URL = 'https://us-central1-oproxy-dev.cloudfunctions.net'  # disable-secrets-detection
```
Comment (glicht) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.py]: Fix to proper prod url. Use same method as we do with graph to configure custom url via auth_id and `@`. See: https://github.com/demisto/content/blob/master/Integrations/MicrosoftGraphMail/MicrosoftGraphMail.py#L26

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.py [hunk 2]
```diff
@@ -0,0 +1,909 @@
...
+        valid_until = integration_context.get('valid_until')
+        if access_token and valid_until:
+            if self.epoch_seconds() < valid_until:
+                return access_token
+
+        body = json.dumps({'app_name': 'google',
+                          'registration_id': AUTH_ID,
+                           'encrypted_token': self.get_encrypted(REFRESH_TOKEN, ENC_KEY)})
+
+        h = httplib2.Http(disable_ssl_certificate_validation=not DISABLE_SSL)
```
Comment (glicht) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.py]: Missing proxy support

**Comment by glicht** [comment]
Let's try to get the playbook test to work. Should run against both oproxy dev and prod as we do with graph.

**Comment by adi88d** [comment]
Done, now we have test for dev and prod

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.yml [hunk 3]
```diff
@@ -0,0 +1,209 @@
+category: Authentication
```
Comment (glicht) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.yml]: Change category to: `Messaging`

```suggestion
category: Messaging
```

**Comment by adi88d** [comment]
@kirbles19 please review

**Comment by richardbluestone** [comment]
Docs reviewed

**Comment by glicht** [comment]
@adi88d let's wait with the merge. I looked a bit at the desktop app flow and it looks like this can be done without oproxy. I will check this and update.

**Comment by glicht** [comment]
Closing this as we will use #5109 for now. We can revisit as we gain experience.

**Pull request closed without merging**